central/complianceoperator/v2/checkresults/service/service_impl.go

```diff
@@ -205,3 +219,74 @@ func (s *serviceImpl) GetComplianceScanCheckResult(ctx context.Context, req *v2.
 
 	return storagetov2.ComplianceV2CheckResult(scanResult), nil
 }
+
+// GetComplianceScanConfigurationResults retrieves the most recent compliance operator scan results for the specified query
+// TODO(ROX-20333):  the most recent portion will come when this ticket is worked once everything is wired up so we can tell
+// what the latest scan is.
+func (s *serviceImpl) GetComplianceScanConfigurationResults(ctx context.Context, request *v2.ComplianceScanResultsRequest) (*v2.ListComplianceScanResultsResponse, error) {
+	if request.GetScanConfigName() == "" {
+		return nil, errors.Wrap(errox.InvalidArgs, "Scan configuration name is required")
+	}
+
+	// Fill in Query.
+	parsedQuery, err := search.ParseQuery(request.GetQuery().GetQuery(), search.MatchAllIfEmpty())
+	if err != nil {
+		return nil, errors.Wrapf(errox.InvalidArgs, "Unable to parse query %v", err)
+	}
+
+	// Fill in pagination.
+	paginated.FillPaginationV2(parsedQuery, request.GetQuery().GetPagination(), maxPaginationLimit)
+
+	// Add the scan config name as an exact match
+	parsedQuery = search.ConjunctionQuery(
+		search.NewQueryBuilder().AddExactMatches(search.ComplianceOperatorScanConfigName, request.GetScanConfigName()).ProtoQuery(),
+		parsedQuery,
+	)
+
+	scanResults, err := s.complianceResultsDS.SearchComplianceCheckResults(ctx, parsedQuery)
+	if err != nil {
+		return nil, errors.Wrapf(errox.InvalidArgs, "Unable to retrieve compliance scan results for request %v", request)
+	}
+
+	// Need to look up the scan config IDs to return with the results.
+	scanConfigToIDs := make(map[string]string, len(scanResults))
```

Comment:
Duplicate code?

Reply:
yeah duplicate to code in `GetComplianceScanResults`